### PR TITLE
Add support for different raster data types in Create constant raster layer algorithm

### PR DESCRIPTION
**Pull request by root676**
## Description
This PR proposes an additional advanced **OUTPUT_TYPE** parameter for the _Create constant raster layer_ algorithm which lets the user select the data type of the output constant raster. Before the algorithm was limited to only create constant _float32_ rasters. Now, raster datasets of different types can be created:
- Byte
- Integer16
- Unsigned Integer16
- Integer32
- Unsigned Integer32
- Float32
- Float64

By default, the **OUTPUT_TYPE** is preset to the default value **float32** in order to prevent corruption of existing models/scripts.
 
In general, this feature enhances the data creation capabilities for constant rasters in QGIS. The PR also includes a **native test** for the algorithm as well as a new **icon**.

Please feel free to comment/review/enhance!

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmconstantraster.cpp [hunk 1]
```diff
@@ -48,7 +50,10 @@ QString QgsConstantRasterAlgorithm::groupId() const
 QString QgsConstantRasterAlgorithm::shortHelpString() const
 {
   return QObject::tr( "Generates raster layer for given extent and cell "
-                      "size filled with the specified value." );
+                      "size filled with the specified value.\n"
+                      "Additionally an output data type can be specified. "
+                      "The algorithm will stop if a value has been entered that "
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmconstantraster.cpp:55]:
```suggestion
                      "The algorithm will abort if a value has been entered that "
```

File: src/analysis/processing/qgsalgorithmconstantraster.cpp [hunk 2]
```diff
@@ -65,6 +70,20 @@ void QgsConstantRasterAlgorithm::initAlgorithm( const QVariantMap & )
   addParameter( new QgsProcessingParameterNumber( QStringLiteral( "NUMBER" ), QObject::tr( "Constant value" ),
                 QgsProcessingParameterNumber::Double, 1, false ) );
 
+  QStringList rasterDataTypes = QStringList(); //currently supported raster data types that can be handled QgsRasterBlock::writeValue()
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmconstantraster.cpp:73]:
```suggestion
  QStringList rasterDataTypes; //currently supported raster data types that can be handled QgsRasterBlock::writeValue()
```

**Review by nyalldawson — COMMENTED** [review]
Looks great, nice improvement!

File: src/analysis/processing/qgsalgorithmconstantraster.cpp (hunk 1 above)
Comment (root676) [reply src/analysis/processing/qgsalgorithmconstantraster.cpp]: fixed

File: src/analysis/processing/qgsalgorithmconstantraster.cpp (hunk 2 above)
Comment (root676) [reply src/analysis/processing/qgsalgorithmconstantraster.cpp]: fixed

**Comment by root676** [comment]
@nyalldawson thanks for the quick review!

**Pull request merged**